third_party/terraform/tests/resource_compute_instance_test.go.erb

```diff
@@ -1950,6 +1954,140 @@ func testAccCheckComputeInstanceDestroyProducer(t *testing.T) func(s *terraform.
+			NetworkBefore:           "a",
+			SubnetworkBefore:        "a",
+			SubnetworkProjectBefore: "a",
+			NetworkIPBefore:         "a",
+
+			NetworkAfter:           "a",
+			SubnetworkAfter:        "a",
+			SubnetworkProjectAfter: "a",
+			NetworkIPAfter:         "b",
+		},
+		"NetworkIP and Network change": {
+			ExpectedForceNew:        false,
+			NetworkBefore:           "a",
+			SubnetworkBefore:        "a",
+			SubnetworkProjectBefore: "a",
+			NetworkIPBefore:         "a",
+
+			NetworkAfter:           "b",
+			SubnetworkAfter:        "a",
+			SubnetworkProjectAfter: "a",
+			NetworkIPAfter:         "b",
+		},
+		"NetworkIP and Subnetwork change": {
+			ExpectedForceNew:        false,
+			NetworkBefore:           "a",
+			SubnetworkBefore:        "a",
+			SubnetworkProjectBefore: "a",
+			NetworkIPBefore:         "a",
+
+			NetworkAfter:           "a",
+			SubnetworkAfter:        "b",
+			SubnetworkProjectAfter: "a",
+			NetworkIPAfter:         "b",
+		},
+		"NetworkIP and SubnetworkProject change": {
+			ExpectedForceNew:        false,
+			NetworkBefore:           "a",
+			SubnetworkBefore:        "a",
+			SubnetworkProjectBefore: "a",
+			NetworkIPBefore:         "a",
+
+			NetworkAfter:           "a",
+			SubnetworkAfter:        "a",
+			SubnetworkProjectAfter: "b",
+			NetworkIPAfter:         "b",
+		},
+		"All change ": {
+			ExpectedForceNew:        false,
+			NetworkBefore:           "a",
+			SubnetworkBefore:        "a",
+			SubnetworkProjectBefore: "a",
+			NetworkIPBefore:         "a",
+
+			NetworkAfter:           "b",
+			SubnetworkAfter:        "b",
+			SubnetworkProjectAfter: "b",
+			NetworkIPAfter:         "b",
+		},
+		"No change": {
+			ExpectedForceNew:        false,
+			NetworkBefore:           "a",
+			SubnetworkBefore:        "a",
+			SubnetworkProjectBefore: "a",
+			NetworkIPBefore:         "a",
+
+			NetworkAfter:           "a",
+			SubnetworkAfter:        "a",
+			SubnetworkProjectAfter: "a",
+			NetworkIPAfter:         "a",
+		},
+	}
+
+	for tn, tc := range cases {
+		d := &ResourceDiffMock{
+			Before: map[string]interface{}{
+				"network_interface.#":                    1,
+				"network_interface.0.network":            tc.NetworkBefore,
+				"network_interface.0.subnetwork":         tc.SubnetworkBefore,
+				"network_interface.0.subnetwork_project": tc.SubnetworkProjectBefore,
+				"network_interface.0.network_ip":         tc.NetworkIPBefore,
+			},
+			After: map[string]interface{}{
+				"network_interface.#":                    1,
+				"network_interface.0.network":            tc.NetworkAfter,
+				"network_interface.0.subnetwork":         tc.SubnetworkAfter,
+				"network_interface.0.subnetwork_project": tc.SubnetworkProjectAfter,
+				"network_interface.0.network_ip":         tc.NetworkIPAfter,
+			},
+		}
+		err := forceNewIfNetworkIPNotUpdatableFunc(d);
+		if  err != nil {
+			t.Errorf(err.Error())
+		}
+		if tc.ExpectedForceNew != d.IsForceNew {
+			t.Errorf("%v: Scenario got unexpected result", tn)
```

Comment:
even though it's easy to look it up, it doesn't hurt to explicitly call out "expected ForceNew to be %v but was %v"

Reply:
good call!